hooks/hookstage/mutation.go

```diff
@@ -0,0 +1,52 @@
+package hookstage
+
+type MutationType int
+
+const (
+	MutationUpdate MutationType = iota
+	MutationDelete
```

Comment:
Were you thinking that an update would also handle create so that it behaved like an upsert operation?

In the PRD I see the following requirement:
`If PBS-core cannot apply a module's requested change (e.g. another module already deleted the json path), it should log a warning condition that will be displayed in debug mode.`
If update is effectively an upsert, then the only error I see occurring is if two modules both intend to delete the same field. Whichever module attempts the deletion last would result in a warning because there was nothing to delete.

Reply:
At the moment, modules are allowed to handle edge cases on their own.
The current implementation of mutations allows modules to define their own behaviour for the operations because, under the hood, mutations use callbacks created by modules.
So the module can verify if a value exists in the payload before updating it, otherwise, it can return an error, which will be displayed in debug mode. Similarly, return an error trying to delete an already deleted path.

But this functionality can be changed with the evolution of the `ChangeSet` concept proposed by Scott. So that `ChangeSet` types can define multiple operations (create/update/upsert/delete) each with its own behaviour.

I think as we start developing the module we will see more use cases and update this functionality.